stratum/lib/security/credentials_manager.cc

```diff
@@ -67,13 +67,11 @@ ::util::Status CredentialsManager::Initialize(bool secure_only) {
       server_credentials_ = ::grpc::InsecureServerCredentials();
     }
   } else {
-    // Verify that the certificate files exist and can be read
+    // Verify that the certificate files exist and are regular (non-symlink) files
     // If files are not present or not accesible, method will return with nullptr
-    std::ifstream ifile1, ifile2, ifile3;
-    ifile1.open(FLAGS_server_key_file);
-    ifile2.open(FLAGS_server_cert_file);
-    ifile3.open(FLAGS_ca_cert_file);
-    if(ifile1 && ifile2 && ifile3) {
+    if (stratum::hal::IsRegularFile(FLAGS_ca_cert_file) &&
```

Comment:
Is it really necessary to qualify the names with `stratum::hal::`? This isn't done for other references to functions in the `utils` module. (My apologies; I forgot to mention this in my original comment.)

I found the following in the same source file:
```
status.Update(WriteStringToFile(root_certs, FLAGS_ca_cert_file));
```

Reply:
There is a `utils` in `namespace stratum` which is where the `WriteStringToFile()` comes from (in stratum/lib/utils.cc)

The `IsRegularFile()` was added to stratum/hal/lib/common/utils.cc which is in `stratum::hal` namespace. Hence the qualifier here. I could have added a `using namespace stratum::hal` at the top of this file, but used qualifiers here instead

Reply:
Oh, the _other_ utils file. I keep forgetting that Stratum has two of them.

I'd have add a `using` myself, but I'm not going to quibble over it.